### ISSUE TITLE
required: True not working

### ISSUE THREAD

**Issue by twigs67** [nestjs/mongoose#638]
Hello,

I'm getting this error:
```
TypeError: Invalid schema configuration: `True` is not a valid type at path `metrics.required`
```
Seems to be the same issue as [this ticket](https://github.com/nestjs/mongoose/issues/397).

```
 "dependencies": {
    "@nestjs/common": "^7.5.1",
    "@nestjs/core": "^7.5.1",
    "@nestjs/platform-express": "^7.5.1",
    "reflect-metadata": "^0.1.13",
    "rimraf": "^3.0.2",
    "rxjs": "^6.6.3",
    "@nestjs/config": "0.5.0",
    "@nestjs/jwt": "^7.0.0",
    "@nestjs/mongoose": "^7.0.2",
    "@nestjs/passport": "^7.0.0",
    "@nestjs/serve-static": "^2.1.3",
    "@nestjs/swagger": "^4.7.0",
    "@types/mongoose": "^5.7.37",
    "class-validator": "^0.12.2",
    "dotenv": "^8.2.0",
    "jwk-to-pem": "^2.0.3",
    "mongoose": "^5.10.13",
    "mysql": "^2.18.1",
    "nanoid": "^3.1.9",
    "nestjs-typegoose": "^7.1.38",
    "passport": "^0.4.1",
    "passport-jwt": "^4.0.0",
    "swagger-ui-express": "^4.1.4"
  },
  "devDependencies": {
    "@nestjs/cli": "^7.5.1",
    "@nestjs/schematics": "^7.1.3",
    "@nestjs/testing": "^7.5.1",
    "@types/express": "^4.17.8",
    "@types/jest": "^26.0.15",
    "@types/node": "^14.14.6",
    "@types/supertest": "^2.0.10",
    "@typescript-eslint/eslint-plugin": "^4.6.1",
    "@typescript-eslint/parser": "^4.6.1",
    "eslint": "^7.12.1",
    "eslint-config-prettier": "^6.15.0",
    "eslint-plugin-prettier": "^3.1.4",
    "jest": "^26.6.3",
    "prettier": "^2.1.2",
    "supertest": "^6.0.0",
    "ts-jest": "^26.4.3",
    "ts-loader": "^8.0.8",
    "ts-node": "^9.0.0",
    "tsconfig-paths": "^3.9.0",
    "typescript": "^4.0.5",
    "@types/dotenv": "^8.2.0",
    "@types/jwk-to-pem": "^2.0.0",
    "@types/ws": "^7.2.5",
    "aws-sdk": "^2.695.0",
    "tslint": "^5.20.0"
  },
  "jest": {
    "moduleFileExtensions": [
      "js",
      "json",
      "ts"
    ],
    "rootDir": "src",
    "testRegex": ".spec.ts$",
    "transform": {
      "^.+\\.(t|j)s$": "ts-jest"
    },
    "coverageDirectory": "./coverage",
    "testEnvironment": "node"
  }
```

**Comment by kamilmysliwiec** [comment]
Please provide a minimum reproduction repository.

**Comment by twigs67** [comment]
@kamilmysliwiec Here is a test repo that reproduces the error - [TEST REPO](https://github.com/twigs67/mongoose-test)

If you need anything else, please let me know.

**Comment by twigs67** [comment]
It was just brought to my attention, that it only breaks when the type is object. For example, this works:
```
    @Prop({
        required: true
    })
    user_id: string;
```

This does not:
```
    @Prop({
        required: true
    })
    metrics: object;
```

**Comment by kamilmysliwiec** [comment]
Fixed in 7.0.3
